Error while running testsuite (simd_lane, simd_load) "loop not vectorized"

- wabt 1.0.34
- clang 17.0.6

Extracted lines:
```
out/test/wasm2c/spec/simd_lane/simd_lane.0.c:967:6: error: loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Werror,-Wpass-failed=transform-warning]
out/test/wasm2c/spec/simd_load/simd_load.12.c:803:6: error: loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Werror,-Wpass-failed=transform-warning]
```

Full build log: [wabt-1.0.34_broken_simd_tests:20240319-155457.log](https://github.com/WebAssembly/wabt/files/14653567/wabt-1.0.34_broken_simd_tests.20240319-155457.log)

--- a/test/run-spec-wasm2c.py
+++ b/test/run-spec-wasm2c.py
@@ -471,6 +471,9 @@ def Compile(cc, c_filename, out_dir, use_c11, *cflags):
                  '-Wno-ignored-optimization-argument',
                  '-Wno-tautological-constant-out-of-range-compare',
                  '-Wno-infinite-recursion',
+                 # simde calls for pragma clang loop vectorize(enable),
+                 # which can generate warnings if the optimization pass fails.
+                 '-Wno-pass-failed',
                  '-fno-optimize-sibling-calls',
                  '-frounding-math', '-fsignaling-nans',
                  '-D_DEFAULT_SOURCE']

